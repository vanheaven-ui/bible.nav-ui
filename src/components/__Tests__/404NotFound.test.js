import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import '@testing-library/jest-dom';
import NotFound from '../404NotFound';

it('renders without crashing', () => {
  ReactDOM.render(<NotFound />, rootDiv());
});

it('does not change unexpectedly', () => {
  const tree = renderer.create(
    <NotFound />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the correct wrapper element', () => {
  const { container } = render(
    <NotFound />,
  );
  expect(container.firstChild.classList.contains('not-found')).toBe(true);
});
