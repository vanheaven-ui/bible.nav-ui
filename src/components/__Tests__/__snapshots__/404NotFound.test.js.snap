// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not change unexpectedly 1`] = `
<section
  className="not-found"
>
  <div
    className="overlay-notfound"
  />
</section>
`;
